tests/store/artifact/test_databricks_model_artifact_repo.py

```diff
@@ -0,0 +1,151 @@
+import pytest
+from unittest import mock
+import json
+from unittest.mock import ANY
+from mlflow.exceptions import MlflowException
+from mlflow.store.artifact.databricks_model_artifact_repo import DatabricksModelArtifactRepository
+
+DATABRICKS_MODEL_ARTIFACT_REPOSITORY_PACKAGE = (
+    "mlflow.store.artifact.databricks_model_artifact_repo"
+)
+DATABRICKS_MODEL_ARTIFACT_REPOSITORY = (
+    DATABRICKS_MODEL_ARTIFACT_REPOSITORY_PACKAGE + ".DatabricksModelArtifactRepository"
+)
+MOCK_MODEL_ROOT_URI_WITH_PROFILE = "models://profile@databricks/MyModel/12"
+MOCK_MODEL_ROOT_URI_WITHOUT_PROFILE = "models:/MyModel/12"
+MOCK_MODEL_NAME = "MyModel"
+MOCK_MODEL_VERSION = "12"
+
+REGISTRY_LIST_ARTIFACTS_ENDPOINT = "/api/2.0/mlflow/model-versions/list-artifacts"
+REGISTRY_ARTIFACT_PRESIGNED_URI_ENDPOINT = "/api/2.0/mlflow/model-versions/get-signed-download-uri"
+
+
+@pytest.fixture()
+def host_creds_mock():
+    with mock.patch(
+        DATABRICKS_MODEL_ARTIFACT_REPOSITORY_PACKAGE + ".get_databricks_host_creds"
+    ) as get_creds_mock:
+        get_creds_mock.return_value = None
+        yield
+
+
+@pytest.fixture()
+def databricks_model_artifact_repo(host_creds_mock):  # pylint: disable=unused-argument
+    return DatabricksModelArtifactRepository(MOCK_MODEL_ROOT_URI_WITH_PROFILE)
+
+
+class TestDatabricksModelArtifactRepository(object):
+    def test_init_validation_and_cleaning(self):
+        with mock.patch(
+            DATABRICKS_MODEL_ARTIFACT_REPOSITORY_PACKAGE + ".get_databricks_host_creds"
+        ) as get_creds_mock:
+            get_creds_mock.return_value = None
+            repo = DatabricksModelArtifactRepository(MOCK_MODEL_ROOT_URI_WITH_PROFILE)
+            assert repo.artifact_uri == MOCK_MODEL_ROOT_URI_WITH_PROFILE
+            assert repo.model_name == MOCK_MODEL_NAME
+            assert repo.model_version == MOCK_MODEL_VERSION
+
+        with pytest.raises(MlflowException):
+            DatabricksModelArtifactRepository("s3://test")
+        with pytest.raises(MlflowException):
+            DatabricksModelArtifactRepository("dbfs:/databricks/mlflow/EXP/RUN/artifact")
+        with pytest.raises(MlflowException):
+            DatabricksModelArtifactRepository(
+                "dbfs://scope:key@notdatabricks/databricks/mlflow-tracking/experiment/1/run/2"
+            )
+        with pytest.raises(MlflowException):
+            DatabricksModelArtifactRepository("models:/MyModel/12")
+        with pytest.raises(MlflowException):
+            DatabricksModelArtifactRepository("models://scope:key@notdatabricks/MyModel/12")
+
+    def test_init_when_profile_is_inferred(self):
+        with mock.patch(
+            DATABRICKS_MODEL_ARTIFACT_REPOSITORY_PACKAGE + ".get_databricks_host_creds"
+        ) as get_creds_mock, mock.patch(
+            "mlflow.tracking.get_registry_uri", return_value="databricks://getRegistryUriDefault"
+        ), mock.patch(
+            DATABRICKS_MODEL_ARTIFACT_REPOSITORY_PACKAGE + ".is_databricks_profile",
```

Comment:
you probably don't want to mock this since get_registry_uri is mocked (though you may need to mock `get_registry_uri` under a different package name?)